packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -0,0 +1,222 @@
+ */
+export class ProxyContext implements EditableTreeContext {
+    public readonly withCursors: Set<ProxyTarget | ProxyTargetSequence> = new Set();
+    public readonly withAnchors: Set<ProxyTarget | ProxyTargetSequence> = new Set();
+    private readonly observers: Dependent[] = [];
+    private readonly afterHandlers: Set<EditableTreeContextHandler> = new Set();
+    private readonly nodes: Map<ETreeNodePath, ProxyTarget | ProxyTargetSequence> = new Map();
+    private emptyNode?: ProxyTarget;
+
+    constructor(
+        public readonly forest: IEditableForest,
+        public readonly tree?: ISharedTree,
+    ) {
+        const observer = new SimpleObservingDependent((token?: InvalidationToken, delta?: Delta.Root): void => {
+            if (token?.isSecondaryInvalidation) {
+                // TODO: this is only an example of after change processing, which is not supported yet by ObjectForest/SharedTree.
+                this.handleAfterChange();
+            } else {
+                this.prepareForEdit();
+            }
+        });
+        this.observers.push(observer);
+        forest.registerDependent(observer);
+    }
+
+    public prepareForEdit(): void {
+        for (const target of this.withCursors) {
+            target.prepareForEdit();
+        }
+        assert(this.withCursors.size === 0, 0x3c0 /* prepareForEdit should remove all cursors */);
+    }
+
+    public free(): void {
+        for (const target of this.withCursors) {
+            target.free();
+        }
+        for (const target of this.withAnchors) {
+            target.free();
+        }
+        for (const observer of this.observers) {
+            this.forest.removeDependent(observer);
+        }
+        assert(this.withCursors.size === 0, 0x3c1 /* free should remove all cursors */);
+        assert(this.withAnchors.size === 0, 0x3c2 /* free should remove all anchors */);
+    }
+
+    public get root(): UnwrappedEditableField {
+        const cursor = this.forest.allocateCursor();
+        const destination = this.forest.root(this.forest.rootField);
+        const cursorResult = this.forest.tryMoveCursorTo(destination, cursor);
+        const targets: ProxyTarget[] = [];
+        if (cursorResult === TreeNavigationResult.Ok) {
+            this.emptyNode?.free();
+            this.emptyNode = undefined;
+            do {
+                targets.push(this.createTarget(cursor));
+            } while (cursor.seek(1) === TreeNavigationResult.Ok);
+        }
+        cursor.free();
+        this.forest.anchors.forget(destination);
+        const rootSchema = lookupGlobalFieldSchema(this.forest.schema, rootFieldKey);
+        const fieldKind = getFieldKind(rootSchema);
+        if (targets.length === 0 && fieldKind.multiplicity !== Multiplicity.Sequence) {
+            targets.push(this.createEmptyTarget());
+        }
+        return proxifyField(this, getFieldKind(rootSchema), targets);
+    }
+
+    public createEmptyTarget(): ProxyTarget {
```

Comment:
Can you document at least this function and/or emptyNode. I won't know why it exists, or why its lazily allocated, what happens if you try to edit it, etc.

Note that before this change, evert method on this type had documentation (inherited from the interface). There are a lot more things here now, but not much more documentation wise.